crates/agglayer-types/src/aggchain_proof.rs

```diff
@@ -0,0 +1,29 @@
+use agglayer_primitives::Signature;
+use serde::{Deserialize, Serialize};
+use sp1_core_machine::reduce::SP1ReduceProof;
+use sp1_prover::InnerSC;
+
+use crate::Digest;
+
+// Aggchain data submitted via the [`Certificate`].
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(untagged)]
+pub enum AggchainData {
```

Comment:
do we want to also rename the file/module to `aggchain_data.rs`?

Reply:
I'm fine keeping this that way, we can rename if we feel that it makes more sense later.